scalding-beam/src/test/scala/com/twitter/scalding/beam_backend/BeamBackendTests.scala

```diff
@@ -388,6 +389,29 @@ class BeamBackendTests extends FunSuite with BeforeAndAfter {
     )
   }
 
+  test("toIterableExecutionTest1") {
+    val input = Seq(5, 3, 2, 6, 1, 4)
+    val bmode = BeamMode.default(pipelineOptions)
+
+    val iter = TypedPipe.from(input).toIterableExecution.waitFor(Config.empty, bmode).get
+    assert(iter.toSet == input.toSet)
+  }
+
+  test("toIterableExecutionWithJoin") {
+    val bmode = BeamMode.default(pipelineOptions)
+    val tp1 = TypedPipe.from(1 to 10).map(x => (x, 1))
+    val tp2 = TypedPipe.from(1 to 10).map(x => (x, 2))
+    val output = tp1
+      .join(tp2)
+      .map(x => (x._1, x._2._1 + x._2._2))
```

Comment:
nit: `.mapValues { case (left, right) => left + right }` would be a bit more idiomatic (always prefer operating only on values if you can since you open up more opportunities for dag optimization.